frontend/app/.server/domain/services/api-client.ts

```diff
@@ -52,24 +61,53 @@ async function baseFetch(path: string, context: string): Promise<Result<Response
 }
 
 export const apiClient = {
-  /**
-   * Fetches data from a given path and parses it as JSON.
-   * @param T The expected type of the JSON response.
-   * @param path The API endpoint path.
-   * @param context A descriptive string for the action, used in error messages.
-   * @returns A Promise resolving to a Result containing the typed data (T) or an AppError.
-   */
-  async get<T>(path: string, context: string): Promise<Result<T, AppError>> {
-    // Get the raw response using our base fetcher
-    const responseResult = await baseFetch(path, context);
+  async get<T>(path: string, context: string, token?: string): Promise<Result<T, AppError>> {
+    const responseResult = await baseFetch(path, context, 'GET', token);
+    if (responseResult.isErr()) {
+      return responseResult;
+    }
 
+    const response = responseResult.unwrap();
+    try {
+      const data = (await response.json()) as T;
+      return Ok(data);
+    } catch (parsingError) {
+      return Err(
+        new AppError(
+          `Failed to parse JSON response on '${context.toLowerCase()}': ${String(parsingError)}`,
+          ErrorCodes.VACMAN_API_ERROR,
+        ),
+      );
+    }
+  },
+
+  async post<T>(path: string, context: string, token?: string, model?: T): Promise<Result<T, AppError>> {
```

Comment:
Request and response data type can be different. POST request requires a body.

```suggestion
  async post<TRequestData, TResponseData = unknown>(path: string, context: string, data: TRequestData, token?: string): Promise<Result<TResponseData , AppError>> {
```